### PR TITLE
CI: Require GPU and OpenMP support

### PR DESCRIPTION
**Pull request by havogt**
After the last upgrade, some tests were silently disabled on jenkins because CUDA wasn't detected anymore.

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
This should make all plans red. Before some were just silently disabling GPU and passing.

**Comment by havogt** [comment]
All plans are red as expected. Now we should merge #1553 and then this PR should become green as well.

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

File: CMakeLists.txt [hunk 1]
```diff
@@ -118,6 +117,14 @@ if(BUILD_TESTING)
     include(detect_test_features)
     detect_c_compiler()
     detect_fortran_compiler()
+    option(GT_TESTS_REQUIRE_OpenMP "Require OpenMP support." OFF)
+    if(GT_TESTS_REQUIRE_OpenMP)
+        find_package(OpenMP COMPONENTS CXX REQUIRED)
+    endif()
+    option(GT_TESTS_REQUIRE_GPU "Require GPU support." OFF)
+    if(GT_TESTS_REQUIRE_GPU AND NOT TARGET stencil_gpu)
```
Comment (havogt) [inline CMakeLists.txt:125]: this is a pragmatic test...

**Review by anstaf — APPROVED** [review]

**Pull request merged**